core/clingutils/test/CMakeLists.txt

```diff
@@ -0,0 +1,34 @@
+# Copyright (C) 1995-2019, Rene Brun and Fons Rademakers.
+# All rights reserved.
+#
+# For the licensing terms see $ROOTSYS/LICENSE.
+# For the list of contributors see $ROOTSYS/README/CREDITS.
+
+# Make llvm and clang available here as we are mostly testing methods directly
+# depending on them.
+include_directories(SYSTEM
+  ${CLANG_INCLUDE_DIRS}
+  ${LLVM_INCLUDE_DIRS}
+  ${CLAD_INCLUDE_DIRS}
+)
+
+include_directories(
+  ../res
+  ../../clingutils/res
+  ../../foundation/res
+  ${CLING_INCLUDE_DIRS}
+  )
+
+set(LLVM_DEPS LLVMSupport)
+if(NOT builtin_clang)
+
+  # FIXME: Passing only LLVMSupport when we are using an external clang does not
+  # expand the transitive dependencies. For instance LLVMSupport should expand
+  # to LLVMSupport LLVMDemangle and optionally curses.
+  # For a unknown reasons using clingInterpreter enables such expansion...
+  set(LLVM_DEPS clingInterpreter)
+  link_directories("${LLVM_LIBRARY_DIR}")
+endif()
```

Comment:
Isn't this already done a directory up? The calls to `include_directories()` and `link_directories()` seem unnecessary here. Also, the reason why it works when not using external llvm/clang is that LLVMSupport, etc, are target names, not just library names, and CMake knows which are the dependencies. Please fix this instead of using a hack to get dependencies.

Reply:
I added this because it did not work, then I needed to add `${CLING_LIBRARIES}` which may have made the section obsolete. I can delete it and add it later if needed because atm my -Dbuiltin_clang=Off setup is broken for some unrelated reason.